src/Compilers/CSharp/Test/Semantic/Semantics/OutVarTests.cs

```diff
@@ -26947,6 +26947,40 @@ public static bool TakeOutParam<T>(T y, out T x)
             Assert.Null(model.GetAliasInfo(x1Decl.Type()));
         }
 
+        [Fact, WorkItem(14717, "https://github.com/dotnet/roslyn/issues/14717")]
+        public void ExpressionVariableInCase()
```

Comment:
I believe this is in scope, this is part of the testing for the fix. Also, there are helpers that make SemanticModel  testing both for out variables and pattern variables very easy.